admin/services/check-start.service.js

```diff
@@ -1,46 +1,100 @@
 'use strict'
 
-const uuidv4 = require('uuid/v4')
 const moment = require('moment')
-const checkFormService = require('../services/check-form.service')
-// const checkWindowDataService = require('../services/data-access/check-window.data.service')
+const R = require('ramda')
+const winston = require('winston')
+
 const checkDataService = require('../services/data-access/check.data.service')
+const checkFormDataService = require('../services/data-access/check-form.data.service')
+const checkFormService = require('../services/check-form.service')
 const checkWindowDataService = require('../services/data-access/check-window.data.service')
+const pinGenerationService = require('../services/pin-generation.service')
+const pupilDataService = require('../services/data-access/pupil.data.service')
+
+const checkStartService = {}
+
+/**
+ * Create a check entry for a pupil, generate a pin, and allocate a form
+ * Called from the admin app when the teacher generates a pin
+ * @param pupilIds
+ * @return {Promise<void>}
+ */
+checkStartService.prepareCheck = async function (pupilIds, dfeNumber) {
+  // TODO: add transaction wrapper around the service calls to generate pins and checks
+
+  if (!dfeNumber) {
+    throw new Error('dfeNumber is required')
+  }
+
+  // Validate the incoming pupil list to ensure that the pupils are real ids
+  // and that they belong to the user's school
+  const pupils = await pupilDataService.sqlFindByIdAndDfeNumber(pupilIds, dfeNumber)
+  const providedSet = new Set(pupilIds)
+  const databaseSet = new Set(pupils.map(p => p.id.toString())) // the incoming form params are stringified numbers
+  const difference = new Set([...providedSet].filter(x => !databaseSet.has(x)))
```

Comment:
What's the case of this going wrong?
Only case I could think of is if someone modifies the pupilIds that are being passed from UI.

Reply:
If that's the case then this should be reusable in other areas where pupil ids are being submitted.

Reply:
Yes, it could be generic.  But do we need it yet?

Reply:
Unless you want to refactor this to a separate method later I believe yes.
Sounds like it should be extracted as a separate service method under pupil service?

Reply:
It's not pupil specific - it needs:--

param1: array of stringified numbers which is the incoming param list
param2: array of objects with Ids from the database to check against

Therefore this could be more generic: e.g. `setValidationService` perhaps? I'm open to better suggestions.

We also have to decide if want to accept the stringified versions, or wether the duty is now on the caller to convert to numbers, as the database is storing Ints for the `ids`?

Reply:
Yea it does sounds like a better approach.
I am guessing the caller would convert to numbers.